pkg/services/cleanup/cleanup.go

```diff
@@ -183,3 +187,15 @@ func (srv *CleanUpService) deleteStaleShortURLs(ctx context.Context) {
 		srv.log.Debug("Deleted short urls", "rows affected", cmd.NumDeleted)
 	}
 }
+
+func (srv *CleanUpService) deleteStaleQueryHistory(ctx context.Context) {
+	// Delete query history from 14+ days ago with exception of starred queries
```

Comment:
I think we should have an limit for stared queries as well. A number higher then we ever expect is better than nothing :)

lets say an max of 200k stared queries?

Reply:
I agree! With query history we wanted to do 100 000k limit and do the cleanup where we delete everything above 100,000. I have 2 questions:
1. Do you think 100,000 limit make sense, or should it be 200,000 in the same way as starred queries
2. Do you think removing queries above limit in cleanup makes sense? E.g. if there is 105,000 queries, in cleanup (which is run every 10 min) we remove 5k queries. Or should have a hard limit and throw an error and not let users add any more queries. 
3. The same question for the stared queries. While it is kind-of okay to remove old normal queries, I am not sure if it is a good experience if we started with removing starred queries.

Reply:
I'd like to address this in separate PR after we figure out the details. So I am resolving this discussion here, but let's definitely talk about this!

Reply:
1. 100k or 200k doesn't matter that much. What ever you think is reasonable :)
2. For entities such as query history or annotations I don't think we should reject creation of new objects. So trimming it down to X every 10 min makes sense in this case. 
3. It's not ideal to remove stared queries but it I think it would be very rare that it happens. The potential downsides to not having limits might be performance problems with the database which is way worse. It doesn't feel likely today. But in a few years when people have been using this feature for some time it might be.